### ISSUE TITLE
When OpenRVDAS services restart, CachedDataWriter hangs

### ISSUE THREAD

**Issue by davidpablocohn** [OceanDataTools/openrvdas#241]
**Describe the bug**

**To Reproduce**
1. ``supervisorctl restart all``
2. look at log for, e.g. ``tail -f /var/log/openrvdas/gyr1.stderr``; it will emit
```
2020-08-28 02:47:40,356.356Z 20 INFO logger_runner.py:127 Starting logger gyr1 config gyr1->file/net
2020-08-28 02:47:44,044.044Z 20 INFO cached_data_writer.py:79 CachedDataWriter trying to connect to localhost:8766
```
and hang.

3. If you first start openrvdas:cached_data_server, then start openrvdas:logger_manager, it will behave normally:
```
2020-08-28 02:49:50,138.138Z 20 INFO logger_runner.py:127 Starting logger gyr1 config gyr1->file/net
2020-08-28 02:49:53,742.742Z 20 INFO cached_data_writer.py:79 CachedDataWriter trying to connect to localhost:8766
2020-08-28 02:49:53,786.786Z 20 INFO listener.py:65 Instantiating gyr1->file/net logger
2020-08-28 02:49:53,787.787Z 20 INFO listener.py:93 Running gyr1->file/net
2020-08-28 02:49:53,851.851Z 20 INFO logfile_writer.py:95 LogfileWriter opening new file: /var/tmp/log/gyr1/raw/NBP1406_gyr1-2020-08-28
```

Adding a delay at the start of logger_manager.py doesn't appear to help.

Changing cruise mode (e.g.   to off, then back to a logging mode) via the web console at :8000 also appears to fix the problem.

**Comment by davidpablocohn** [comment]
Turns out it wasn't the CachedDataWriter - it was InfluxDBWriter bailing out with a RuntimeError if it couldn't connect to InfluxDB on the first try. InfluxDB takes a while to spin up, so the logger manager would be ready before it was. Added a loop for InfluxDBWriter to sleep, then try again.

**Comment by davidpablocohn** [comment]
Resolved with 3188af6..be33ce6
